dev-docs/design/how-lit-works.md

```diff
@@ -0,0 +1,320 @@
+
+A comment that matches an expression marker ([added in 2.1](#21-join-the-templates-string-literals-with-a-marker-string)), marks a `TemplatePart` in ChildPart position.
+
+## 3. Create
+
+The create phase is performed when rendering for the the first time to a specific container or Part.
+
+<blockquote>
+
+Note: A template can be rendered to a container (with `render(result, container)`) or a Part. Templates are rendered to Parts by being used as a value within another template.
+
+In this example:
+
+```js
+let postBodyTemplate = (text) => html`<p>${text}</p>`;
+let postTemplate = (post) => html`
+  <h1>${post.title}</h1>
+  ${postBodyTemplate(post.body)}
+`;
+```
+
+The `TemplateResult` returned from `postBodyTemplate(post.body)` is rendered to the `ChildPart` defined by the expression markers.
+
+</blockquote>
+
+<blockquote>
+
+Note: When a `TemplateResult` is rendered to a container or Part we get its `Template` via a `_$getTemplate` which is responsible for creating and caching Template objects.
+
+Certain extensions to lit-html may override `_$getTemplate` to modify templates or cache based on additional keys. `shady-render.js` library uses `ShadyCSS` (part of the Shadow DOM polyfills) to modify templates for CSS scoping, and caches templates based on their scope identifier in addition to their template strings.
+
+`_$getTemplate` is what initiates the [Prepare step](#2-prepare) if a `Template` is not yet in the cache.
+
+</blockquote>
+
+### 3.1 Create a `TemplateInstance`
+
+`TemplateInstance` is responsible for creating the initial DOM and updating that DOM. It's an updatable instance of a `Template`. The `TemplateInstance` holds references to the Parts used to update the DOM.
+
+### 3.2 Clone the template
+
+The `<template>` element is cloned by `TemplateInstance#_clone()`.
+
+Care must be taken to manage cloning and upgrade order. We need the cloned DOM to exactly match the template DOM so that we can find Part locations when we walk the cloned tree. When Custom Elements upgrade, it's possible that they could modify the DOM before we get a chance to walk it, throwing off our part indices.
+
+In general, Custom Elements are barred from modifying their own DOM, and we don't need to worry about their ShadowRoots, except in the case of the Web Components polyfills. So with native Web Components, template contents are cloned with `document.importNode()`, which will cause Custom Elements to upgrade. With polyfilled Web Components, template contents are cloned with `template.content.cloneNode()`, which will not upgrade elements. They are then upgraded later.
```

Comment:
Great comment! Because this document is about `How lit-html works` and not how the polyfills work - I think I am going to cut this paragraph that was copied over from the wiki.